### PR TITLE
feat: referendum wallet votes

### LINKED ISSUE #1788: OpenGov. Referendum. Votes
As a user I want to see my votes

Todo:
- [x] Voted section with total amount of votes
- [x] "My votes" modal window with fragmentation of account/votes/voting power

[Figma link](https://www.figma.com/design/ylvl7Ox25UuZj73Rn85DK7/OpenGov?node-id=574-100471&t=RafEPHCQCBylgBPp-4)
[My votes popup](https://www.figma.com/design/ylvl7Ox25UuZj73Rn85DK7/OpenGov-voting?node-id=273-27249&t=hZlnKtCMb35nXJZl-4)

### PR DESCRIPTION
**Pull request by johnthecat**
* Implemented current wallet votes dialog
  Pop up (1)
* Moved voting history implementation to on chain data
* Refactored governance types
* Bump typescript
* Added pre-commit hooks
* Added eslint boundary plugin for later

### CODE REVIEW

File: src/renderer/entities/governance/lib/claimScheduleService.ts [hunk 1]
```diff
@@ -144,29 +139,29 @@ function gapClaimableLock({ trackId, voting, gap, currentBlockNumber }: GapLockP
...
 ): ClaimableLock[] {
-  const priorLock: ClaimableLock = {
-    claimAt: { type: 'at', block: voting.casting.prior.unlockAt } as ClaimTime,
+  const priorLock = {
+    claimAt: { type: 'at', block: voting.casting.prior.unlockAt },
     amount: voting.casting.prior.amount,
-    affected: [{ trackId, type: 'track' } as AffectTrack],
-  };
+    affected: [{ trackId, type: 'track' }],
+  } satisfies ClaimableLock;
```
Comment (sokolova-an) [inline src/renderer/entities/governance/lib/claimScheduleService.ts:162]: this is a great thing, we should use more now!

File: src/renderer/features/governance/aggregates/walletVotes.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { votingAggregate } from './voting';
+
+export const walletVotesAggregate = {
+  $votes: votingAggregate.$activeWalletVotes,
+};
```
Comment (sokolova-an) [inline src/renderer/features/governance/aggregates/walletVotes.ts]: do we really need this aggregator or will you add more fields there in the future?

File: src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx [hunk 3]
```diff
@@ -88,15 +105,22 @@ export const ReferendumDetailsDialog = ({ chain, referendum, onClose }: Props) =
...
-              {t('governance.referendum.advanced')}
-            </Button>
-          </DetailsCard>
+          {referendumService.isOngoing(referendum) && votingAsset ? (
+            <DetailsCard>
+              <Button className="p-0 h-auto w-fit" size="sm" variant="text" onClick={() => setShowAdvanced(true)}>
+                {t('governance.referendum.advanced')}
+              </Button>
+            </DetailsCard>
+          ) : null}
```
Comment (sokolova-an) [inline src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx]: maybe 
`{referendumService.isOngoing(referendum) && votingAsset && (<DetailsCard> ...)}`

**Review by sokolova-an — APPROVED** [review]

File: src/renderer/entities/governance/lib/lockService.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { type BN, BN_ZERO, bnMax } from '@polkadot/util';
+
+import { type ClaimTime, type ClaimTimeAt, type ClaimTimeUntil } from '@shared/api/governance';
+import { type Conviction, type Voting } from '@shared/core';
+
+import { votingService } from './votingService';
+
+const lockPeriod = {
```
Comment (Asmadek) [inline src/renderer/entities/governance/lib/lockService.ts]: Better to name enums with PascalCase

File: src/renderer/entities/governance/lib/votingService.ts [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
   type Tally,
-  type TrackId,
   type Voting,
-  VotingType,
+  type VotingMap,
 } from '@/shared/core';
+import { toKeysRecord } from '@shared/lib/utils';
 
-const getVoteFractions = (tally: Tally, approve: BN): Record<'aye' | 'nay' | 'pass', number> => {
+const convictionMultipliers: Record<Conviction, number> = {
```
Comment (Asmadek) [inline src/renderer/entities/governance/lib/votingService.ts]: Same enum

**Review by Asmadek — APPROVED** [review]

File: src/renderer/features/governance/aggregates/walletVotes.ts (hunk 2 above)
Comment (johnthecat) [reply src/renderer/features/governance/aggregates/walletVotes.ts]: Yes, I think it's unnecessary

File: src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx (hunk 3 above)
Comment (johnthecat) [reply src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx]: Than check will look more like
```tsx
{referendumService.isOngoing(referendum) && !!votingAsset && (
  <DetailsCard>
)}
```

**Pull request merged**